lib/parser.js

```diff
@@ -22,13 +24,25 @@ function message (commitText) {
 
   // <summary> <newline>* <body>
   if (isNewline(scanner.peek())) {
+    // TODO(@byCedric): include <newline>* in AST.
     while (isNewline(scanner.peek())) {
       scanner.next()
     }
   } else {
     throw scanner.abort(node)
   }
-  node.children.push(body(scanner))
+  const b = body(scanner)
+  if (!(b instanceof Error)) {
+    node.children.push(b)
+  }
+  // TODO(@byCedric): include <newline>* in AST.
+  while (isNewline(scanner.peek())) {
+    scanner.next()
+  }
+  while (!scanner.eof()) {
```

Comment:
What happens if I format my message like this:

```
feat: some nice feature

Biep boep

Ref #1337
Author: @someone

o shit, forgot this
```

Right now, we don't output a failure, we just ignore it. Maybe we should either:
- Abort if additional text was provided and throw an error
- Add another `body` with text below the footer, even though it's not correct

My reason for this is possible typos in the footer. If you take the example above, but instead of `Ref #1337`, im using `Ref#1337`, it's just ignored. Or even something like `Ref JIRA-1337`, which isn't parsed as footer.

Reply:
@byCedric today, this would put the whole:

```
Biep boep

Ref #1337
Author: @someone

o shit, forgot this
```

Into the `body`:

```
message[2] (1:1-8:20, 0-83)
├─0 summary[4] (1:1-1:24, 0-23)
│   ├─0 type "feat" (1:1-1:5, 0-4)
│   ├─1 separator ":" (1:5-1:6, 4-5)
│   ├─2 whitespace " " (1:6-1:7, 5-6)
│   └─3 text "some nice feature" (1:7-1:24, 6-23)
└─1 body[6] (4:1-8:20, 35-83)
    ├─0 text "Biep boep" (3:1-3:10, 25-34)
    ├─1 text "" (4:1-4:1, 35-35)
    ├─2 text "Ref #1337" (5:1-5:10, 36-45)
    ├─3 text "Author: @someone" (6:1-6:17, 46-62)
    ├─4 text "" (7:1-7:1, 63-63)
    └─5 text "o shit, forgot this" (8:1-8:20, 64-83)
```

This isn't ideal, but my concern with being too permissive ab out footers, is a body like this:

```
feat: add new nodes to parser

This PR introduces several new nodes to the
parser: body, footer, newlines. This is the first
in several commits.
```

👆 `parser:` looks like  footer, but is part of the body.